Framework/Core/include/Framework/AnalysisHelpers.h

```diff
@@ -124,11 +124,11 @@ struct OutputForTable {
 /// means of the WritingCursor helper class, from which produces actually
 /// derives.
 template <typename T>
-struct Produces : WritingCursor<typename soa::PackToTable<typename T::table_t::persistent_columns_t>::table> {
+requires(!std::is_same_v<void, typename aod::MetadataTrait<T>::metadata>) struct Produces : WritingCursor<typename soa::PackToTable<aod::MetadataTrait<T>::metadata::origin(), typename T::table_t::persistent_columns_t>::table> {
```

Comment:
Why the "is_same_v"? shouldn't this simply be a concept for traits with metadata?

Reply:
We have a bunch of traits to test for metadata, being an iterator or a table, etc. I plan to do a second pass replacing those with concepts in one go.